src/tyr/route_serializer_osrm.cc

```diff
@@ -1494,65 +1503,94 @@ json::ArrayPtr voice_instructions(const valhalla::DirectionsLeg::Maneuver* prev_
+    // right at the beginning of the navigation. This is something like:
+    // Drive West on XYZ Street.
+    // This voice_instruction_start is only created once. It is always played, even when
+    // the maneuver would otherwise be too short.
+    json::MapPtr voice_instruction_start = json::map({});
+    voice_instruction_start->emplace("distanceAlongGeometry", json::fixed_t{distance, 1});
+    voice_instruction_start->emplace("announcement",
+                                      prev_maneuver->verbal_pre_transition_instruction());
+    voice_instruction_start->emplace("ssmlAnnouncement",
+                                      "<speak>" + prev_maneuver->verbal_pre_transition_instruction() + "</speak>");                                 
+    voice_instructions_array->emplace_back(std::move(voice_instruction_start));
+  } else if (
+    distance_before_verbal_transition_alert_instruction > 1.0 &&
+    distance > distance_before_verbal_transition_alert_instruction +
+                            APPROXIMATE_VERBAL_POSTRANSITION_LENGTH &&
+              !prev_maneuver->verbal_post_transition_instruction().empty()) {
+    // In all other cases we want to play the verbal_post_transition_instruction shortly
+    // after the maneuver has started but only if there is sufficient time to play. On 
+    // the one hand distance_before_verbal_transition_alert_instruction has to be set, 
+    // so there is enough time to play the transition instruction afterwards. On the other hand
+    // there has to be enough time to play the upcoming verbal_pre_transition_instruction and 
+    // the verbal_post_transition_instruction itself. 
+    // The approximation here is that the verbal_post_transition_instruction takes 100
+    // meters to play + the 10 meters after the maneuver start which is added so that the
+    // instruction is not played directly on the intersection where the maneuver starts.
+    json::MapPtr voice_instruction_beginning = json::map({});
+    voice_instruction_beginning->emplace("distanceAlongGeometry", json::fixed_t{distance - 10, 1});
+    voice_instruction_beginning->emplace("announcement",
+                                          prev_maneuver->verbal_post_transition_instruction());
+    voice_instruction_beginning->emplace("ssmlAnnouncement",
+                                      "<speak>" + prev_maneuver->verbal_post_transition_instruction() + "</speak>"); 
+    voice_instructions_array->emplace_back(std::move(voice_instruction_beginning));
+  }
+
+  // If there is an alert instruction and we have enough time to play it, we will play it
+  // TODO: don't add alert instruction if there was already a pre/post transition aleart which is close by
+  if (!maneuver.verbal_transition_alert_instruction().empty() 
+    && distance_before_verbal_transition_alert_instruction > -1.0
+    && prev_maneuver->time() > SECONDS_BEFORE_VERBAL_TRANSITION_ALERT_INSTRUCTION + APPROXIMATE_VERBAL_PRERANSITION_SECONDS) {
```

Comment:
This is one of the most important changes. We will not have the alert instructions for short maneuvers. Doesn't make sense in my mind to alert something on a 50 meter road. Just play the transition instruction in that case is fine enough. Two instructions on that small maneuver are way too much. 
Actually we should even go further and collapse multiple maneuver into a single one if the maneuver itself is too short